junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ExtensionRegistry.java

```diff
@@ -88,28 +90,47 @@ private ExtensionRegistry(ExtensionRegistry parent) {
 	}
 
 	/**
-	 * Get all {@code Extensions} in this registry or one of its ancestors of
+	 * Stream all {@code Extensions} in this registry or one of its ancestors of
 	 * the specified type.
 	 *
 	 * @param extensionType the type of {@link Extension} to get
 	 * @see #getReversedExtensions(Class)
-	 * @see #stream(Class)
+	 * @see #getExtensions(Class)
 	 */
-	public <E extends Extension> List<E> getExtensions(Class<E> extensionType) {
-		List<E> extensions = new ArrayList<>();
-
-		if (this.parent != null) {
-			extensions.addAll(this.parent.getExtensions(extensionType));
+	public <E extends Extension> Stream<E> streamExtensions(Class<E> extensionType) {
```

Comment:
> The _thing_ in question is an extension registry. Thus getting a stream from it will almost certainly stream the extensions contained in that registry.

Yes but the same argument could be made for `getExtensions`. I was just irked by the fact that their names differed more than their intent and think they should be unified. But whether it's `get`/`stream`, `getExtensions`/`streamExtensions`, or `getExtensions`/`getExtensionsAsStream` is a secondary concern (although I like the PR's variant best).

But let's not get distracted by bikeshedding. The much more "important" (well, relatively) change is that delegation order changed. I guess that part is ok?